lib/workers/repository/update/pr/index.ts

```diff
@@ -82,16 +138,33 @@ export async function ensurePr(
   );
 
   const config: BranchConfig = { ...prConfig };
-
+  const filteredPrConfig = generatePrFingerprintConfig(config);
+  const prFingerprint = fingerprint(filteredPrConfig);
   logger.trace({ config }, 'ensurePr');
   // If there is a group, it will use the config of the first upgrade in the array
   const { branchName, ignoreTests, prTitle = '', upgrades } = config;
   const dependencyDashboardCheck =
     config.dependencyDashboardChecks?.[config.branchName];
-  // Check if existing PR exists
+  // Check if PR already exists
   const existingPr = await platform.getBranchPr(branchName);
   if (existingPr) {
     logger.debug('Found existing PR');
+    const prCache = getPrCache(branchName);
+    if (prCache) {
+      logger.trace({ prCache }, 'Found existing PR cache');
+      // return if pr cache is valid and pr was not changed in the past 24hrs
+      if (validatePrCache(prCache, prFingerprint)) {
+        logger.debug(
+          'PR cache matches and no PR changes in last 24hrs, so skipping PR body check'
+        );
+        return { type: 'with-pr', pr: existingPr };
+      }
+    } else {
+      if (config.repositoryCache === 'enabled') {
+        logger.debug('PR cache not found, creating new');
+      }
+      setPrCache(branchName, prFingerprint);
```

Comment:
Are you sure you can validly set pr cache at this time? You know there's a PR open but you don't know anything about what it contains

Reply:
Yes setting it here will be a mistake. It will cause problem when updates get cancelled. Thanks